### PR TITLE
[RPD-147] Add command to force unlock

### PR DESCRIPTION
**Pull request by dudeperf3ct**
Depending on implementation of RPD-146,  we implement  a new command `matcha force-unlock` to forcefully remove lock on remote matcha state on azure storage.

TODO

* [x] Merge PR #122 
* [x] Update function definitions and conflicts once merged

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [x] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [x] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/core/core.py [hunk 1]
```diff
@@ -61,3 +61,9 @@ def analytics_opt_out() -> None:
 def analytics_opt_in() -> None:
     """Enable the collection of anonymous usage data (enabled by default)."""
     GlobalParameters().analytics_opt_out = False
+
+
+def unlock_state_lock() -> None:
```
Comment (Christopher-Norman) [inline src/matcha_ml/core/core.py:66]: Is `remove_state_lock` a better name?

File: src/matcha_ml/cli/cli.py [hunk 2]
```diff
@@ -126,6 +126,15 @@ def destroy(
         remote_state_manager.deprovision_state_storage()
 
 
+@app.command()
+def force_unlock() -> None:
+    """Force unlock lock on remote matcha state on azure."""
```
Comment (Christopher-Norman) [inline src/matcha_ml/cli/cli.py:131]: NIT: """Force unlock remote matcha state on Azure"""

File: tests/test_core/test_core.py [hunk 3]
```diff
@@ -201,3 +208,39 @@ def test_opt_in_subcommand(
...
+
+    # Exit code 0 means there was no error
+    assert result.exit_code == 0
+
+    # Check if unlock function from RemoteStateManager class is called
+    mock_unlock.assert_called_once()
+
+
+def test_unlock_state_lock_not_called(runner, mock_unlock):
+    """Test if unlock function in RemoteStateManager is not called.
```
Comment (Christopher-Norman) [inline tests/test_core/test_core.py:232]: `"""Test if unlock function in RemoteStateManager is not called when running the user does not confirm.`

**Review by Christopher-Norman — CHANGES_REQUESTED** [review]

File: tests/test_core/test_core.py (hunk 3 above)
Comment (Christopher-Norman) [reply tests/test_core/test_core.py]: My mistake this doesn't make sense `"""Test that the unlock function is not called when the user does not confirm.`, does this sound better?

File: tests/test_core/test_core.py [hunk 4]
```diff
@@ -201,3 +208,39 @@ def test_opt_in_subcommand(
...
+
+    # Exit code 0 means there was no error
+    assert result.exit_code == 0
+
+    # Check if unlock function from RemoteStateManager class is called
+    mock_unlock.assert_called_once()
+
+
+def test_remove_state_lock_not_called(runner, mock_unlock):
+    """Test if unlock function in RemoteStateManager is not called when running the user does not confirm.
```
Comment (Christopher-Norman) [inline tests/test_core/test_core.py:232]: My mistake, I think `"""Test that the unlock function is not called when a user runs matcha force-unlock and does not confirm.` is possibly better?

File: tests/test_cli/test_force_unlock.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        runner (CliRunner): typer CLI runner
+    """
+    # Invoke force-unlock command with help option
+    result = runner.invoke(app, ["force-unlock", "--help"])
+
+    # Exit code 0 means there was no error
+    assert result.exit_code == 0
+
+    # Assert string is present in cli output
+    assert "Force unlock remote matcha state on Azure." in result.stdout
```
Comment (Christopher-Norman) [inline tests/test_cli/test_force_unlock.py:23]: Failing test due to the updated docstring in force unlock command

Comment (dudeperf3ct) [reply tests/test_cli/test_force_unlock.py:23]: Updated thanks!

File: tests/test_core/test_core.py (hunk 4 above)
Comment (dudeperf3ct) [reply tests/test_core/test_core.py]: Updated

**Review by Christopher-Norman — APPROVED** [review]
LGTM, nice work

File: tests/test_core/test_core.py [hunk 6]
```diff
@@ -201,3 +208,39 @@ def test_opt_in_subcommand(
     # Check the contents of the config file match
     with open(config_file_path) as f:
         assert dict(yaml.safe_load(f)) == expected_configuration
+
+
+def test_unlock_state_lock_called(runner, mock_unlock):
```
Comment (osw282) [inline tests/test_core/test_core.py:213]: Do we need these tests in `test_core.py`? Feels more like a cli test.

Maybe mocking the result of `self.azure_storage.blob_exists` and test for the status message is more suitable in here? Open to any suggestions or ideas.

**Review by osw282 — COMMENTED** [review]
Nice work. Just one question. 👍🏻

File: tests/test_core/test_core.py (hunk 6 above)
Comment (dudeperf3ct) [reply tests/test_core/test_core.py]: You are right, these feel like cli tests. I have modified `test_core` to test `remove_state_lock` function as it is what is getting implemented in `core.py`. 

Good point about testing for  `self.azure_storage.blob_exists`, will add tests for this.

**Review by osw282 — APPROVED** [review]
Nice tests, LGTM! 💯

**Pull request merged**